server/src/internalClusterTest/java/org/opensearch/recovery/RecoveryWhileUnderLoadIT.java

```diff
@@ -476,7 +493,7 @@ private void logSearchResponse(int numberOfShards, long numberOfDocs, int iterat
 
     private void refreshAndAssert() throws Exception {
         assertBusy(() -> {
-            RefreshResponse actionGet = client().admin().indices().prepareRefresh().get();
+            RefreshResponse actionGet = refreshAndWaitForReplication();
```

Comment:
what is `assertAllSuccessful` doing here?  If its ensuring all reqs are ack'd, can we move the refreshAndWaitForReplication after this assertBusy? Or have two helper methods here - one that refreshes and waits and one that just waits? then we can add the latter after this assertBusy and not nest.

Reply:
Yes makes sense. I created two helper methods with refresh `refreshAndWaitForReplication()`, another without refresh `waitForReplication()`